setup/module/api/user_data.sh

```diff
@@ -1,9 +1,99 @@
+Content-Type: multipart/mixed; boundary="==BOUNDARY=="
+MIME-Version: 1.0
+
+--==BOUNDARY==
 #!/bin/bash
+exec > >(tee /var/log/user-data.log|logger -t user-data -s 2>/dev/console) 2>&1
 echo ECS_CLUSTER=${cluster_id} >> /etc/ecs/ecs.config
 echo ECS_ENGINE_AUTH_TYPE=dockercfg >> /etc/ecs/ecs.config
-yum install -y aws-cli awslogs
+echo ECS_LOGLEVEL=info >> /etc/ecs/ecs.config
+
+# Known issue where packages will sometimes not install on the first try
```

Comment:
After briefly reading the following doc https://docs.aws.amazon.com/AmazonECS/latest/developerguide/docker-debug-mode.html , I'm curious how much of this script needs to be included when we're already using the AWS ECS optimized AMI. 

Also curious where the source of much of this script is located (unless you wrote the majority of it?)

Reply:
The script largely came from this page: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/using_cloudwatch_logs.html

Reply:
The problem with just enabling debug mode in docker is that the logs aren't being shipped anywhere for review. The page I linked talks about how you have to leverage the awslogs Log Driver to actually send logs to CloudWatch, so I modified it a bit to suit our needs.

Reply:
Couple thoughts:

- We should reference in a comment where the source of that script came from. 
- I wonder if we're doubling up commands given that we're already using the ECS Optimized AMI? It's hard to tell whether or not the entirety of that script is required.

Reply:
I'll add a comment for where I got the script from.

As for the contents, in the documentation, it actually makes a few explicit references that the configurations in the script are to be used in ECS-optimized instances:
```
You can configure your container instances to send log information to CloudWatch Logs. This enables you to view different logs from your container instances in one convenient location. This topic helps you get started using CloudWatch Logs on your container instances that were launched with the Amazon ECS-optimized AMI.
```

```
The CloudWatch Logs agent configuration file (/etc/awslogs/awslogs.conf) describes the log files to send to CloudWatch Logs. The agent configuration file's [general] section defines common configurations that apply to all log streams, and you can add individual log stream sections for each file on your container instances that you want to monitor. For more information, see CloudWatch Logs Agent Reference in the Amazon CloudWatch User Guide.

The example configuration file below is configured for the Amazon ECS-optimized AMI, and it provides log streams for several common log files:
```

So it looks to me like you are supposed to include these configurations and things when launching an ECS-optimized AMI.